### PR TITLE
repart: Switch to new mkfs.btrfs subvolume API

### PR DESCRIPTION
**Pull request by daandemeyer**
In https://github.com/kdave/btrfs-progs/pull/877 the API is changing
to "--subvol <path>:ro,default" so let's adapt our usage to match.

This also adds support for read-only subvolumes.

Fixes https://github.com/systemd/systemd/issues/34134

### CODE REVIEW

File: src/partition/repart.c
Comment (yuwata) [inline src/partition/repart.c:270]: `assert(s);`

File: src/partition/repart.c
Comment (yuwata) [inline src/partition/repart.c]: Unnecessary empty line.

File: src/partition/repart.c
Comment (yuwata) [inline src/partition/repart.c]: Indentation.

**Review by yuwata — APPROVED** [review]
LGTM. Several minor comments.

**Comment by DaanDeMeyer** [comment]
Let's not merge this until the corresponding PR in btrfs-progs is merged

File: man/repart.d.xml
Comment (poettering) [inline man/repart.d.xml]: hmm, this is weird. This is not really a flag, it's a singleton after all. kinda weird exposing it like this

Comment (daandemeyer) [reply man/repart.d.xml]: OK we can stick with the previous solution for the default subvolume then.

File: src/shared/image-policy.c
Comment (poettering) [inline src/shared/image-policy.c:440]: commit msg is misleading. not a null byte, but a null pointer.

File: src/partition/repart.c
Comment (poettering) [inline src/partition/repart.c:2247]: i wonder if it wouldn't be nicer to simply add the default subvol to the existing subvolumes if missing, with default settings. but doesn't matter

File: src/partition/repart.c
Comment (poettering) [inline src/partition/repart.c]: add a helper for this?

Comment (poettering) [reply src/partition/repart.c]: you have the same loop twice after all

**Comment by DaanDeMeyer** [comment]
The corresponding PR in mkfs.btrfs was merged, so merging this as well

**Pull request merged**
